app/src/main/java/com/zeapo/pwdstore/pwgen/PasswordGenerator.kt

```diff
@@ -7,139 +7,131 @@ package com.zeapo.pwdstore.pwgen
-     * <tr><td>A</td><td>don't include uppercase letters</td></tr>
-     * <tr><td>B</td><td>don't include ambiguous charactersl</td></tr>
-     * <tr><td>s</td><td>generate completely random passwords</td></tr>
-     * <tr><td>v</td><td>don't include vowels</td></tr>
-     * <tr><td>y</td><td>include at least one symbol</td></tr>
-     * <tr><td>L</td><td>don't include lowercase letters</td></tr>
-    </table> *
-     * @param numArgv numerical options for password generation: length of
-     * generated passwords followed by number of passwords to
-     * generate
-     * @return `false` if a numerical options is invalid,
-     * `true` otherwise
+     * Enables the [PasswordOption]s in [options] and sets [targetLength] as the length for
+     * generated passwords.
      */
-    @JvmStatic
-    fun setPrefs(ctx: Context, argv: ArrayList<String>, vararg numArgv: Int): Boolean {
+    fun setPrefs(ctx: Context, options: List<PasswordOption>, targetLength: Int): Boolean {
         ctx.getSharedPreferences("PasswordGenerator", Context.MODE_PRIVATE).edit {
-            for (option in pwOptions.toCharArray()) {
-                if (argv.contains(option.toString())) {
-                    putBoolean(option.toString(), true)
-                    argv.remove(option.toString())
-                } else {
-                    putBoolean(option.toString(), false)
-                }
-            }
-            var i = 0
-            while (i < numArgv.size && i < 2) {
-                if (numArgv[i] <= 0) {
-                    // Invalid password length or number of passwords
-                    return false
-                }
-                val name = if (i == 0) "length" else "num"
-                putInt(name, numArgv[i])
-                i++
-            }
+            for (possibleOption in PasswordOption.values())
+                putBoolean(possibleOption.key, possibleOption in options)
+            putInt("length", targetLength)
         }
         return true
     }
 
+    fun isValidPassword(password: String, pwFlags: Int): Boolean {
+        if (pwFlags hasFlag DIGITS && password.none { it in DIGITS_STR })
+            return false
+        if (pwFlags hasFlag UPPERS && password.none { it in UPPERS_STR })
+            return false
+        if (pwFlags hasFlag LOWERS && password.none { it in LOWERS_STR })
+            return false
+        if (pwFlags hasFlag SYMBOLS && password.none { it in SYMBOLS_STR })
+            return false
+        if (pwFlags hasFlag NO_AMBIGUOUS && password.any { it in AMBIGUOUS_STR })
+            return false
+        return true
+    }
+
     /**
-     * Generates passwords using the preferences set by
-     * [.setPrefs].
-     *
-     * @param ctx context from which to retrieve SharedPreferences from
-     * preferences file 'PasswordGenerator'
-     * @return list of generated passwords
+     * Generates a password using the preferences set by [.setPrefs].
```

Comment:
```suggestion
     * Generates a password using the preferences set by [setPrefs].
```

I believe it works without the period